src/test/java/FillingTheFormTest.java

```diff
@@ -0,0 +1,176 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+public class FillingTheFormTest {
+
+    Map<String, String> originalDataSet = new HashMap<>()
+    {
+        {
+            put("firstName", "Adam");
+            put("lastName", "Johns");
+            put("emailAddress", "adam.johns@test.com");
+            put("age", "25");
+            put("salary", "25000");
+            put("department", "Test Department");
+        }
+    };
+
+    Map<String, String> updatedDataSet = new HashMap<>()
+    {
+        {
+            put("firstName", "Mary");
+            put("lastName", "Smith");
+            put("emailAddress", "mary.smith@test.com");
+            put("age", "31");
+            put("salary", "150000");
+            put("department", "Test Update Department");
+        }
+    };
+
+
+    @Test (testName = "FillingTheFormTest")
+    public void testFillingTheForm() throws InterruptedException {
+        WebDriver driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        driver.get("https://demoqa.com/webtables");
+
+        //Opening The Form For Adding New Entry
+
+        driver.findElement(By.xpath("//button[@id='addNewRecordButton']")).click();
+
+        // Finding Web Elements On The Form
+
+        WebElement firstName = driver.findElement(By.xpath("//div[@class=\"fade modal show\"]//div[@class=\"modal-content\"]/div[@class=\"modal-body\"]/form//div[@class=\"col-md-6 col-sm-6\"]//input[@id='firstName']"));
```

Comment:
Very long locator. It could be simplified to `//input[@id='firstName']` The same is true for code bellow